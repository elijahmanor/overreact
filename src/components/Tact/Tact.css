.Tact {
  /* margin: 0.5rem; */
  background-color: var(--color-section-bg);
  border-bottom: 1px solid var(--color-divider);
  display: flex;
}

.Tact-avatar {
  border-radius: 50%;
  height: 5rem;
  margin: 1rem;
}

.Tact-body {
  width: 100%;
  margin: 1rem 1rem 1rem 0;
}

.Tact-header {
  display: flex;
  color: var(--color-text-secondary);
}

.Tact-fullName {
  color: var(--color-text-primary);
  font-weight: bold;
}

.Tact-userName {
}

.Tact-message {
  margin-bottom: 0.5rem;
}

.Tact-date {
}

.Tact-comment {

}

.Tact-actions {
  
}

.TactActions-action {
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  outline: inherit;
  margin-right: 1rem;
}

.TactActions-action.is-active .fa-heart {
  color: red;
}
  