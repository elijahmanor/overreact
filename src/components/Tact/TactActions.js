import React from "react";
import PropTypes from "prop-types";

import "./Tact.css";

const TactActions = ({ id, userName, favorite }) => {
  return <div className="TactActions">
    <button className="TactActions-action"><i className="fa fa-comment-o" aria-hidden="true"></i></button>
    <button className="TactActions-action"><i className="fa fa-heart-o" aria-hidden="true"></i></button>
    <button className="TactActions-action is-active"><i className="fa fa-heart" aria-hidden="true"></i></button>
  </div>;
};

TactActions.propTypes = {
  id: PropTypes.number.isRequired,
  userName: PropTypes.string.isRequired,
  favorite: PropTypes.bool.isRequired
};

export default TactActions;
