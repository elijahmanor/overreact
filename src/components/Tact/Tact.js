import React from "react";
import md5 from "md5";
import { format, formatDistanceStrict } from "date-fns";
import PropTypes from "prop-types";
import TactActions from "./TactActions";

import "./Tact.css";

const Tact = ({ id, date, userName, fullName, avatar, email, message, favorite }) => {
  const url = avatar || `https://www.gravatar.com/avatar/${md5(email)}`;

  return <div className="Tact">
    <img className="Tact-avatar" alt={`${fullName}'s Avatar`} src={url} />
    <article className="Tact-body">
      <header className="Tact-header">
        <div className="Tact-fullName">{fullName}</div>
        <span>&nbsp;</span>
        <div className="Tact-userName">@{userName}</div>
        <span>&nbsp;â€”&nbsp;</span>
        <div className="Tact-date" title={format(new Date(date), "LLL")}>{formatDistanceStrict(new Date(date), new Date())}</div>
      </header>
      <div className="Tact-message">{message}</div>
      <TactActions id={id} userName={userName} favorite={favorite} />
    </article>
  </div>;
};

Tact.propTypes = {
  id: PropTypes.number.isRequired,
  date: PropTypes.instanceOf(Date).isRequired,
  userName: PropTypes.string.isRequired,
  fullName: PropTypes.string.isRequired,
  avatar: PropTypes.string,
  email: PropTypes.string,
  message: PropTypes.string.isRequired,
  favorite: PropTypes.bool.isRequired
};

export default Tact;
