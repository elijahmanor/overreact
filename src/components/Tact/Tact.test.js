import React from "react";
import { shallow, mount } from "enzyme";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Tact from "./Tact";

describe("Tact", () => {
  it("renders with an avatar", () => {
    const tree = renderer.create(
      <Tact
        id={1}
        date={new Date()}
        userName="johnsmith"
        fullName="John Smith"
        avatar="http://avatar.com/me.png"
        message="Hello World!"
        favorite={false}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it("renders with an email", () => {
    const tree = renderer.create(
      <Tact
        id={1}
        date={new Date()}
        userName="johnsmith"
        fullName="John Smith"
        avatar="http://avatar.com/me.png"
        message="Hello World!"
        favorite={false}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
