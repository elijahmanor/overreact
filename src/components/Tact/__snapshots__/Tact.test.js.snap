// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tact renders with an avatar 1`] = `
<div
  className="Tact"
>
  <img
    alt="John Smith's Avatar"
    className="Tact-avatar"
    src="http://avatar.com/me.png"
  />
  <article
    className="Tact-body"
  >
    <header
      className="Tact-header"
    >
      <div
        className="Tact-fullName"
      >
        John Smith
      </div>
      <span>
         
      </span>
      <div
        className="Tact-userName"
      >
        @
        johnsmith
      </div>
      <span>
         — 
      </span>
      <div
        className="Tact-date"
        title="October 10 2017 9:58 a.m."
      >
        0 seconds
      </div>
    </header>
    <div
      className="Tact-message"
    >
      Hello World!
    </div>
    <div
      className="TactActions"
    >
      <button
        className="TactActions-action"
      >
        <i
          aria-hidden="true"
          className="fa fa-comment-o"
        />
      </button>
      <button
        className="TactActions-action"
      >
        <i
          aria-hidden="true"
          className="fa fa-heart-o"
        />
      </button>
      <button
        className="TactActions-action is-active"
      >
        <i
          aria-hidden="true"
          className="fa fa-heart"
        />
      </button>
    </div>
  </article>
</div>
`;

exports[`Tact renders with an email 1`] = `
<div
  className="Tact"
>
  <img
    alt="John Smith's Avatar"
    className="Tact-avatar"
    src="http://avatar.com/me.png"
  />
  <article
    className="Tact-body"
  >
    <header
      className="Tact-header"
    >
      <div
        className="Tact-fullName"
      >
        John Smith
      </div>
      <span>
         
      </span>
      <div
        className="Tact-userName"
      >
        @
        johnsmith
      </div>
      <span>
         — 
      </span>
      <div
        className="Tact-date"
        title="October 10 2017 9:58 a.m."
      >
        0 seconds
      </div>
    </header>
    <div
      className="Tact-message"
    >
      Hello World!
    </div>
    <div
      className="TactActions"
    >
      <button
        className="TactActions-action"
      >
        <i
          aria-hidden="true"
          className="fa fa-comment-o"
        />
      </button>
      <button
        className="TactActions-action"
      >
        <i
          aria-hidden="true"
          className="fa fa-heart-o"
        />
      </button>
      <button
        className="TactActions-action is-active"
      >
        <i
          aria-hidden="true"
          className="fa fa-heart"
        />
      </button>
    </div>
  </article>
</div>
`;
